engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java

```diff
@@ -0,0 +1,162 @@
+import org.hibernate.validator.internal.properties.ConstrainableType;
+import org.hibernate.validator.internal.util.Contracts;
+import org.hibernate.validator.internal.util.privilegedactions.GetDeclaredConstructor;
+import org.hibernate.validator.internal.util.privilegedactions.GetDeclaredField;
+import org.hibernate.validator.internal.util.privilegedactions.GetDeclaredMethod;
+import org.hibernate.validator.internal.util.privilegedactions.GetMethod;
+import org.hibernate.validator.spi.properties.ConstrainableExecutable;
+import org.hibernate.validator.spi.properties.GetterPropertySelectionStrategy;
+
+/**
+ * {@link ConstrainableType} that describes a JavaBeanHelper.
+ *
+ * @author Marko Bekhta
+ */
+public class JavaBeanHelper {
+
+	private final GetterPropertySelectionStrategy getterPropertySelectionStrategy;
+
+	public JavaBeanHelper(GetterPropertySelectionStrategy getterPropertySelectionStrategy) {
+		this.getterPropertySelectionStrategy = getterPropertySelectionStrategy;
+	}
+
+	public Optional<JavaBeanField> findDeclaredFieldProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+
+		Field field = run( GetDeclaredField.action( declaringClass, property ) );
+		return Optional.ofNullable( field ).map( JavaBeanField::new );
+	}
+
+	public Optional<JavaBeanGetter> findDeclaredGetterProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+		for ( String possibleName : getterPropertySelectionStrategy.getPossibleMethodNamesForGetterProperty( property ) ) {
+			Method getter = run( GetDeclaredMethod.action( declaringClass, possibleName ) );
+			if ( getter != null ) {
+				return Optional.of( new JavaBeanGetter( declaringClass, getter, getterPropertySelectionStrategy.getPropertyName( new JavaBeanConstrainableExecutable( getter ) ) ) );
+			}
+		}
+		return Optional.empty();
+	}
+
+	public Optional<JavaBeanGetter> findGetterProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+
+		Optional<JavaBeanGetter> declaredGetterProperty = findDeclaredGetterProperty( declaringClass, property );
+
+		if ( declaredGetterProperty.isPresent() ) {
+			return declaredGetterProperty;
+		}
+		else {
+			// look for getters from the superclasses and interfaces
+			for ( String possibleName : getterPropertySelectionStrategy.getPossibleMethodNamesForGetterProperty( property ) ) {
+				Method getter = run( GetMethod.action( declaringClass, possibleName ) );
+				if ( getter != null ) {
+					return Optional.of( new JavaBeanGetter( declaringClass, getter, getterPropertySelectionStrategy.getPropertyName( new JavaBeanConstrainableExecutable( getter ) ) ) );
+				}
+			}
+			return Optional.empty();
+		}
+	}
+
+	public Optional<JavaBeanMethod> findDeclaredMethodByNameAndParameters(Class<?> declaringClass, String methodName, Class<?>... parameterTypes) {
```

Comment:
Just remove the `ByNameAndParameters`. It doesn't bring anything considering the parameters are very clear (and you only have one method anyway).